File: app/src/main/java/com/service/endlessservicewtbroadcastreceiver/utils/RestartBroadcastReceiver.java
package com.service.endlessservicewtbroadcastreceiver.utils;


import static android.content.Context.JOB_SCHEDULER_SERVICE;

import android.app.job.JobInfo;
import android.app.job.JobScheduler;
import android.content.BroadcastReceiver;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class RestartBroadcastReceiver extends BroadcastReceiver {
    private static JobScheduler jobScheduler;

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(RestartBroadcastReceiver.class.getSimpleName(), "Service Stopped, but this is a never ending service.");
        scheduleJob(context);
    }

    public static void scheduleJob(Context context) {
        if (jobScheduler == null) {
            jobScheduler = (JobScheduler) context
                    .getSystemService(JOB_SCHEDULER_SERVICE);
        }
        ComponentName componentName = new ComponentName(context,
                JobService.class);
        JobInfo jobInfo = new JobInfo.Builder(1, componentName)
                .setOverrideDeadline(0)
                .setPersisted(true).build();
        jobScheduler.schedule(jobInfo);
    }
}

File: app/src/main/java/com/service/endlessservicewtbroadcastreceiver/utils/RestartBroadcastReceiver.kt
package com.service.endlessservicewtbroadcastreceiver.utils

import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.content.BroadcastReceiver
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.util.Log

class RestartBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.i(RestartBroadcastReceiver::class.java.simpleName, "Service Stopped, but this is a never ending service.")
        scheduleJob(context)
    }

    companion object {
        private var jobScheduler: JobScheduler? = null
        fun scheduleJob(context: Context) {
            if (jobScheduler == null) {
                jobScheduler = context
                    .getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
            }
            val componentName = ComponentName(
                context,
                JobService::class.java
            )
            val jobInfo = JobInfo.Builder(1, componentName)
                .setOverrideDeadline(0)
                .setPersisted(true).build()
            jobScheduler!!.schedule(jobInfo)
        }
    }
}
